extensions/pkg/controller/utils_test.go

```diff
@@ -270,4 +271,36 @@ var _ = Describe("Utils", func() {
 			Expect(controller.IsMigrated(worker)).To(BeTrue())
 		})
 	})
+
+	Describe("#GetObjectByReference", func() {
+		var (
+			ctx = context.TODO()
+			ref = &autoscalingv1.CrossVersionObjectReference{
+				APIVersion: "v1",
+				Kind:       "Secret",
+				Name:       "foo",
+			}
+			namespace = "shoot--test--foo"
+			refSecret = &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      controller.ReferencedResourcesPrefix + "foo",
+					Namespace: namespace,
+				},
+				Data: map[string][]byte{
+					"foo": []byte("bar"),
+				},
+			}
+		)
+
+		It("will error", func() {
+			secret := &corev1.Secret{}
+			c.EXPECT().Get(ctx, client.ObjectKey{Namespace: namespace, Name: controller.ReferencedResourcesPrefix + ref.Name}, secret).DoAndReturn(
+				func(_ context.Context, _ client.ObjectKey, secret *corev1.Secret) error {
+					*secret = *refSecret
```

Comment:
I think `DeepCopy`/`DeepCopyInto` is probably the cleaner way to copy API objects.
```suggestion
					refSecret.DeepCopyInto(secret)
```